bionic/core/task_execution.py

```diff
@@ -505,7 +571,7 @@ def set_up_caching_flags(self, bootstrap):
             which may cause the values to be subtly different.
             To avoid this warning, disable persistence for the decorators
             by {disable_message}."""
-            logging.warn(message)
+            logger.warn(message)
```

Comment:
This change does not belong in this PR but I just realized we are using `warnings.warn` in some places and `logger.warn` in other. We should make these warnings consistent.